kernel/timer.c

```diff
@@ -126,7 +126,7 @@ void z_impl_k_timer_start(struct k_timer *timer, k_timeout_t duration,
 	 * argument the same way k_sleep() does), but historical.  The
 	 * timer_api test relies on this behavior.
 	 */
-	if (period.ticks != 0 && Z_TICK_ABS(period.ticks) < 0) {
+	if ((K_TIMEOUT_EQ(period, K_FOREVER) == 0) && period.ticks != 0 && Z_TICK_ABS(period.ticks) < 0) {
```

Comment:
This should just be `K_TIMEOUT_EQ(period, K_FOREVER)` without the wrapping parentheses and a comparison to `0`.  That macro evaluates to a boolean expression.

Reply:
Would this be acceptable?
```suggestion
	if (!K_TIMEOUT_EQ(period, K_FOREVER) && period.ticks != 0 && Z_TICK_ABS(period.ticks) < 0) {
```

Reply:
yes